core/src/replay_stage.rs

```diff
@@ -578,6 +580,20 @@ impl ReplayStage {
             let _exit = Finalizer::new(exit.clone());
             let mut identity_keypair = cluster_info.keypair().clone();
             let mut my_pubkey = identity_keypair.pubkey();
+            if my_pubkey != startup_identity {
```

Comment:
So if set-identity comes in after this line could we successfully update it in the loop?

Reply:
yeah since my_pubkey is cached until the condition is checked in the loop we are fine if rpc executes between here and the loop

Reply:
smaller change is to just load `my_pubkey` from `tower.node_pubkey` and let the existing set-identity handle things. But this might be better for quick turnaround